be/src/exprs/runtime_filter_slots.h

```diff
@@ -62,16 +69,42 @@ class RuntimeFilterSlotsBase {
             DCHECK(runtime_filter->expr_order() < _probe_expr_context.size());
 
             // do not create 'in filter' when hash_table size over limit
-            bool over_max_in_num = (hash_table_size >= state->runtime_filter_max_in_num());
+            auto max_in_num = state->runtime_filter_max_in_num();
+            bool over_max_in_num = (hash_table_size >= max_in_num);
 
             bool is_in_filter = (runtime_filter->type() == RuntimeFilterType::IN_FILTER);
 
-            // do not create 'bloom filter' and 'minmax filter' when 'in filter' has created
-            bool pass_not_in = (has_in_filter[runtime_filter->expr_order()] &&
-                                !runtime_filter->has_remote_target());
-
-            if (over_max_in_num == is_in_filter && (is_in_filter || pass_not_in)) {
-                ignore_filter(filter_desc.filter_id);
+            // Note:
+            // In the case that exist *remote target* and in filter and other filter,
+            // we must merge other filter whatever in filter is over the max num in current node,
+            // because:
+            // case 1: (in filter >= max num) in current node, so in filter will be ignored,
+            //         and then other filter can be used
+            // case 2: (in filter < max num) in current node, we don't know whether the in filter
+            //         will be ignored in merge node, so we must transfer other filter to merge node
+            if (!runtime_filter->has_remote_target()) {
+                bool exists_in_filter = has_in_filter[runtime_filter->expr_order()];
+                if (is_in_filter && over_max_in_num) {
+                    LOG(INFO) << "fragment instance " << print_id(state->fragment_instance_id())
+                              << " ignore runtime filter(in filter id " << filter_desc.filter_id
+                              << ") because: in_num(" << hash_table_size
+                              << ") >= max_in_num(" << max_in_num << ")";
+                    ignore_local_filter(filter_desc.filter_id);
+                    continue;
+                } else if (!is_in_filter && exists_in_filter) {
```

Comment:
Don't we need to judge over_max_in_num here? If `in` be ignored, we can't ignore other filter.
Will it be ignore `minmax/bloomfilter` in this case? 
```cpp
is_in_filter=false;
over_max_in_num=true;
exists_in_filter=true;

```

Reply:
okay